momentGW/pbc/ints.py

```diff
@@ -15,6 +16,31 @@
 from momentGW.ints import Integrals, require_compression_metric
 
 
+def require_uncompressed_naux():
+    """
+    Determine the uncompressed number of auxiliary basis functions
+    per k-point before running the function.
+    """
+
+    def decorator(func):
+        @functools.wraps(func)
+        def wrapper(self, *args, **kwargs):
+            if self._naux is None:
+                uncomp_naux = np.zeros(len(self.kpts), dtype=int)
+                for ki in self.kpts.loop(1):
+                    for block in self.with_df.sr_loop((0, ki), compact=False):
+                        if block[2] == -1:
+                            raise NotImplementedError("Low dimensional integrals")
+                        uncomp_naux[ki] += block[0].shape[0]
+
+                self._naux = uncomp_naux
+            return func(self, *args, **kwargs)
+
+        return wrapper
+
+    return decorator
+
+
```

Comment:
As per our discussion, have removed the decorator and have changed the code so that `self.naux_full` behaves as expected with it representing the `naux` values before compression per k-point. A  `self._naux_full` is introduced to store these values so they aren't always calculated each time `self.naux_full` is called.